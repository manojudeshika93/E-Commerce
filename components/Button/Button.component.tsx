import React, { useMemo, useState } from 'react';
import { Pressable, Text, View } from 'react-native';

import { tw } from '@/config';

import { ButtonProps } from './Button.types';

export function Button({ onPress, title, disabled = false, fullWidth = true, isLink = false }: Readonly<ButtonProps>) {
  const [isButtonClicked, setIsButtonClicked] = useState(false);

  const backgroundColor = useMemo(() => {
    if (disabled) return 'bg-secondary-400';
    if (isLink) return 'bg-custom-transparent';
    if (isButtonClicked) return 'bg-primary-400';
    return 'bg-primary-500';
  }, [isButtonClicked, disabled]);

  const pressableStyle = tw.style(`items-center justify-center rounded-2xl h-15 ${backgroundColor}`, { 'w-full': fullWidth });

  return (
    <Pressable
      style={pressableStyle}
      disabled={disabled}
      onPress={onPress}
      onPressIn={() => setIsButtonClicked(true)}
      onPressOut={() => setIsButtonClicked(false)}>
      <View style={tw`items-center justify-center`}>
        {title.length > 0 && <Text style={tw`text-h3-semibold ${isLink ? 'text-secondary-400 underline' : 'text-custom-white'}`}>{title}</Text>}
      </View>
    </Pressable>
  );
}
