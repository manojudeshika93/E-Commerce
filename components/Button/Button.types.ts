export interface ButtonProps {
  onPress?: () => void;
  title: string;
  disabled?: boolean;
  fullWidth?: boolean;
  isLink?: boolean;
}
