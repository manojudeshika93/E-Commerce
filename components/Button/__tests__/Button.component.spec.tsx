import { fireEvent, render } from '@testing-library/react-native';
import React from 'react';

import { Button } from '../Button.component';

describe('Button Component', () => {
  const onPressMock = jest.fn();

  const getRenderedComponent = (props = {}) =>
    render(<Button onPress={onPressMock} title="Test Button" {...props} />);

  it('should render the Button component correctly', () => {
    const { toJSON } = getRenderedComponent();
    expect(toJSON()).toMatchSnapshot();
  });

  it('should display the correct title', () => {
    const { getByText } = getRenderedComponent();
    expect(getByText('Test Button')).toBeTruthy();
  });

  it('should call onPress when button is pressed', () => {
    const { getByText } = getRenderedComponent();
    const buttonElement = getByText('Test Button');

    fireEvent.press(buttonElement);
    expect(onPressMock).toHaveBeenCalled();
  });
});
