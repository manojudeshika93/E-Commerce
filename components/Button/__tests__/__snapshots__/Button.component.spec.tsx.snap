// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should render the Button component correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#C30145",
      "borderRadius": 16,
      "height": 60,
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        {
          "color": "#FFFFFF",
          "fontFamily": "Inter_600SemiBold",
          "fontSize": 20,
          "lineHeight": 30,
        }
      }
    >
      Test Button
    </Text>
  </View>
</View>
`;
